portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,11 +80,56 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty(COMMENT_TEXT, commentText);
     commentEntity.setProperty(TIME_STAMP, timeStamp);
 
+    // TODO: Remove print statement and store image URL in database.
+    System.out.println(getUploadedFileUrl(request, /* formInputElementName= */ "image").get());
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect(REDIRECT_URL);
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload an image file. */
+  private Optional<String> getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (local server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return Optional.empty();
+    }
+
+    // Gets first and only file in form submission.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return Optional.empty();
+    }
+
+    String fileInfo = blobInfo.getContentType();
+
+    // Return null if file is not a jpg, png or tiff image.
+    if (!fileInfo.equals(JPEG) && !fileInfo.equals(PNG) && !fileInfo.equals(TIFF)) {
+      return Optional.empty();
+
+    }
+
+    // Gets URL that points to the uploaded file.
+    ImagesService imagesService = ImagesServiceFactory.getImagesService();
+    ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+
+    // Returns relative path to image.
```

Comment:
This comment explains what the code is doing. You only need comments when it is not clear what is happening from the code. I believe this comment and potentially others can be omitted.

Reply:
fixed